# HG changeset patch
# User horii
# Date 1473905514 14400
# Node ID 8d16f74380a78eb76cb33183a64440316393903e
# Parent  be698ac288484ab140715ee29ed9335e6ea8a33b
8165231: java.nio.Bits.unaligned() doesn't return true on ppc
Reviewed-by: simonis, coffeys

diff -r be698ac28848 -r 8d16f74380a7 src/share/classes/java/nio/Bits.java
--- openjdk/jdk/src/share/classes/java/nio/Bits.java	Fri Sep 23 18:19:23 2016 +0000
+++ openjdk/jdk/src/share/classes/java/nio/Bits.java	Wed Sep 14 22:11:54 2016 -0400
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2000, 2012, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -614,7 +614,8 @@
         String arch = AccessController.doPrivileged(
             new sun.security.action.GetPropertyAction("os.arch"));
         unaligned = arch.equals("i386") || arch.equals("x86")
-            || arch.equals("amd64") || arch.equals("x86_64");
+            || arch.equals("amd64") || arch.equals("x86_64")
+            || arch.equals("ppc64") || arch.equals("ppc64le");
         unalignedKnown = true;
         return unaligned;
     }
diff -r be698ac28848 -r 8d16f74380a7 src/share/classes/sun/security/provider/ByteArrayAccess.java
--- openjdk/jdk/src/share/classes/sun/security/provider/ByteArrayAccess.java	Fri Sep 23 18:19:23 2016 +0000
+++ openjdk/jdk/src/share/classes/sun/security/provider/ByteArrayAccess.java	Wed Sep 14 22:11:54 2016 -0400
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2006, 2013, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2006, 2016, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -94,7 +94,7 @@
         String arch = java.security.AccessController.doPrivileged
             (new sun.security.action.GetPropertyAction("os.arch", ""));
         return arch.equals("i386") || arch.equals("x86") || arch.equals("amd64")
-            || arch.equals("x86_64");
+            || arch.equals("x86_64") || arch.equals("ppc64") || arch.equals("ppc64le");
     }
 
     /**

